python/ray/dataframe/dataframe.py

```diff
@@ -1503,14 +1503,127 @@ def drop_helper(obj, axis, label):
             self._block_partitions = obj._block_partitions
 
     def drop_duplicates(self, subset=None, keep='first', inplace=False):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        """Return DataFrame with duplicate rows removed, optionally only
+        considering certain columns
+
+        Args:
+            subset: column label or sequence of labels, optional
+                Only consider certain columns for identifying duplicates, by
+                default use all of the columns
+            keep: {'first', 'last', False}, default 'first'
+            inplace: Whether to drop duplicates in place or to return a copy
+
+        Returns:
+            deduplicated: DataFrame
+        """
+        duplicated_array = self.duplicated(subset, keep).as_matrix()
+        dropped_rows = self.index.values[duplicated_array]
+        return self.drop(dropped_rows, inplace=inplace)
 
     def duplicated(self, subset=None, keep='first'):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        """Return boolean Series denoting duplicate rows, optionally only
+        considering certain columns
+
+        Args:
+            subset: column label or sequence of labels, optional
+                Only consider certain columns for identifying duplicates, by
+                default use all of the columns
+            keep: {'first', 'last', False}, default 'first'
+
+        Returns:
+            duplicated: Series
+        """
+        pd.DataFrame(columns=self.columns).duplicated(subset=subset, keep=keep)
+        if subset is None:
+            subset = self.columns
+        elif (not np.iterable(subset) or
+              isinstance(subset, compat.string_types) or
+              isinstance(subset, tuple) and subset in self.columns):
+            subset = subset,
```

Comment:
Hanging comma

Reply:
this is to make the subset a tuple. do you think it would be better to write `subset = (subset,)`?

Reply:
`(subset,)` is more idiomatic.